src/store/actions.js

```diff
@@ -14,14 +14,24 @@ import {
   isAccountNotFoundError,
   executeAndSetInterval,
 } from '../popup/utils/helper';
-import { CURRENCIES_URL } from '../popup/utils/constants';
+import { CURRENCIES_URL, MODAL_DEFAULT } from '../popup/utils/constants';
+import { i18n } from './plugins/languages';
 
 export default {
   switchNetwork({ commit }, payload) {
     commit('switchNetwork', payload);
     commit('setMiddleware', null);
     commit('initTransactions');
   },
+
+  async selectNetwork({ dispatch, getters }, network) {
+    await dispatch('switchNetwork', network);
+    if (getters.tippingSupported) return;
+    await dispatch('modals/open', {
+      name: MODAL_DEFAULT,
+      ...i18n.t('modals.tip-mainnet-warning'),
```

Comment:
I'm not sure about this. I think we should pass data to the modal as the object prop values.